charmcraft/application/main.py

```diff
@@ -68,9 +68,19 @@ def _extra_yaml_transform(
     ) -> dict[str, Any]:
         yaml_data = yaml_data.copy()
 
+        ext_names = yaml_data.setdefault("extensions", [])
+        invalid_extensions = [ext for ext in ext_names if ext.startswith("_")]
```

Comment:
is this PR still needed? I think we discussed doing the transformations in extra_yaml_transforms